toolkit/tools/imagecustomizer/docs/verity.md

```diff
@@ -0,0 +1,141 @@
+```
+
+## cloud-init-local, cloud-config, cloud-final
+
+These services require write access to certain directories, such as
+`/var/lib/cloud`, which conflicts with the read-only nature of the
+verity-protected root filesystem.
+
+### Solution: Create a writable persistent partition
+
+To provide the required write access, create a writable persistent partition for
+`/var`. This will allow the `cloud-init-local`, `cloud-config`, and
+`cloud-final` services to function correctly without issues caused by the
+read-only root filesystem.
+
+```yaml
+storage:
+  disks:
+  - partitionTableType: gpt
+    maxSize: 5120M
+    partitions:
+    - id: boot
+      start: 1M
+      end: 1024M
+    - id: root
+      start: 1024M
+      end: 3072M
+    - id: roothash
+      start: 3072M
+      end: 3200M
+    - id: var
+      start: 3200M
+  filesystems:
+  - deviceId: boot
+    type: ext4
+    mountPoint:
+      path: /boot
+  - deviceId: root
+    type: ext4
+    mountPoint:
+      path: /
+  - deviceId: var
+    type: ext4
+    mountPoint:
+      path: /var
+```
+
+## docker
+
+The `docker.service` requires write access to `/var/lib/docker`, which is
+restricted by the read-only root filesystem.
+
+### Solution: Create a writable persistent partition
+
+To unblock `docker.service`, create a writable persistent partition for `/var`.
+This allows Docker to write to `/var/lib/docker` without any issues caused by
+the read-only root filesystem. 
+
+Refer to the YAML layout under the
+[cloud-init](#cloud-init-local-cloud-config-cloud-final) section for the
+partition configuration.
+
+## auditd, logrotate
+
+The `auditd.service` and `logrotate.service` need write access to `/var/log` for
+logging purposes. However, the root filesystem's read-only nature blocks these
+services.
+
+### Solution: Create a writable persistent partition
+
+To unblock `auditd.service` and `logrotate.service`, ensure that `/var` is
+mounted as a writable persistent partition, allowing write access to `/var/log`.
+
+Refer to the YAML layout under the
+[cloud-init](#cloud-init-local-cloud-config-cloud-final) section for the
+partition configuration.
+
+## sshd
+
+The `sshd` service requires write access to the SSH host keys, which by default
+are stored in `/etc/ssh`. However, with the root filesystem being read-only,
+this prevents `sshd` from running correctly.
+
+### Solution: Create a writable persistent partition and redirect SSH host keys
+
+To resolve this, create a writable partition for `/var` and redirect the SSH
+host keys from `/etc` to `/var`. This ensures that `sshd` can write and access
+the necessary keys without encountering issues due to the read-only root
+filesystem.
+
+Refer to the YAML layout under the
+[cloud-init](#cloud-init-local-cloud-config-cloud-final) section for the
+partition configuration.
+
+The following example script can be added to the `postCustomization` scripts to
+automatically move the SSH host keys and update the `sshd_config` file:
+
+```bash
```

Comment:
The `sshd-keygen.service` file also need to be updated.

Note: Additional files now support `content` instead of `source`.

Reply:
Fixed, thanks.